test/unit_test/CMakeLists.txt

```
@@ -56,6 +57,7 @@ add_executable(vw-unit-test.out
 
 # Add the include directories from vw target for testing
 target_link_libraries(vw-unit-test.out PRIVATE vw_core RapidJSON Boost::unit_test_framework)
+target_include_directories(vw-unit-test.out PRIVATE ${VW_CORE_SOURCE_DIR})
```

Rather than this, add the src directory as part of the include for the core lib and then:
```suggestion
target_include_directories(vw-unit-test.out PRIVATE $<TARGET_PROPERTY:vw_core,INCLUDE_DIRECTORIES>)
```

---

done, let me know if you think it's something we should put into the vw_add_library section